README.md

```diff
@@ -76,6 +86,94 @@ Example:
 ['id', 'name', productCategory: ['id', 'name']]
 ```
 
+---
+
+### Service class inclusion
+
+This approach can be used to `graphqlize` **any** kind of ruby (service) class
+so that it has re-usable graphql query methods.
+
+* First add `extend GraphqlConnector::<server>::Query` in the the class
+* Add a `return_fields` which defines what fields are going to be returned
+* Next for each mapping add a `query` that alias the type that the server supports as follows
+  * `query <alias>: <query type in graphql server>, params: [<any kind of query type params>]`
+  * If query type does not need them, omit `params`
+
+Examples:
+
+```ruby
+GraphqlConnector.configure do |config|
+  config.add_server(name: 'Foo', uri: 'http://foo.com/api/graphql', headers: {})
+end
+
+# product.rb
+class Product
+  extend GraphqlConnector::Foo::Query
+  return_fields :id, :name, product_category: [:id, :name]
+
+  query all: :products_all
+  query by_id: :products_all, params: :id
+  query by_names: :product_all, params: :names
+  query by: :product_all, params: [:id, :name]
+  query by_category_id: :product_all, params: :product_category
+end
+
+Product.all
+=> [OpenStruct<id=1, name='Demo Product', product_category=<ProductCategory<id=10, name='Demo Category'>>, ...]
+
+Product.by_id(id: 1)
+=> OpenStruct<id=1, name='Demo Product', product_category=<ProductCategory<id=10, name='Demo Category'>>
+
+Product.by_names(names: ['Demo Product', 'Non Demo Product'])
+=> [OpenStruct<id=1, name='Demo Product', product_category=<ProductCategory<id=10, name='Demo Category'>>, Product<id=2, name='Demo Product' ...]
+
+Product.by(id: 1, name: 'Demo Product')
+=> OpenStruct<id=1, name='Demo Product'>
+
+Product.by_category_id(product_category: { id: 10})
+=> OpenStruct<id=1, name='Demo Product', product_category=<ProductCategory<id=10, name='Demo Category'>>
+```
+
+Also custom **class methods** can used to call any kind of `query` and do further selection instead:
+
+```ruby
+class Product
+  include GraphqlConnector::Foo::Query
+  return_fields :id, :name, product_category: [:id, :name]
+
+  query all: :products_all
+
+  def self.by_id(id:)
+    all.select { |products| products.id == id }.first
+  end
+end
+
+Product.by_id(id: 1)
+=> OpenStruct<id=1, name='Demo Product', product_category=<ProductCategory<id=10, name='Demo Category'>>
+```
+
+Last but not least raw queries can also be used, like the following:
+
+```ruby
+class Product
+  include GraphqlConnector::Foo::Query
+  return_fields :id, :name, product_category: [:id, :name]
+
+  query all: ' query { products { id name } } '
+  query by: ' query { products($id: !ID, $name: !String) '\
+            '{ products(id: $id, name: $name) { id name } } } ',
+             params: [:id, :name]
+
+end
+
+Product.all
+=> [ { id: '1', name: 'Demo Product', ...]
```

Comment:
I think it should be like this, right:
```ruby
[ { id: '1', name: 'Demo Product' }, ...]
```
Otherwise i would expect the `...` means there are more attributes of one object.

Reply:
Fixed typo